core/perfutil/perfutil.go

```diff
@@ -0,0 +1,252 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package perfutil
+
+import (
+	"fmt"
+	"time"
+)
+
+var (
+	enableTrace bool
+	perfUuid string
+	allTransactionTraces map[string] *traceOperation
+)
+
+type traceOperation struct {
+	name string
+	start int64
+	end int64
+	succ bool
+	next *traceOperation
+}
+
+// set Trace uuid
+func SetPerfUuid(uuid string)  {
+	perfUuid = uuid
+	return
+}
+
+// get Trace uuid
+func GetPerfUuid() (string) {
+	return perfUuid
+}
+
+//Trace state
+func TraceState() (bool) {
+	return enableTrace
+}
+
+//enable/disable trace
+func SetTrace(val bool) {
+	enableTrace = val
+	if val {
+		allTransactionTraces = make(map[string] *traceOperation)
+	} else {
+		allTransactionTraces = nil
+	}
+}
+
+//get the pointer to the end of the link list
+func GetPT(uuid string) *traceOperation {
+	if enableTrace {
+		to := allTransactionTraces[uuid]
+		if to == nil {
+			fmt.Printf("[GetPT] Transaction trace nil. \n")
+			return to
+		}
+		for {
```

Comment:
We want to keep the linked list in the order of tracepoints so that the trace can be printed in the order they are added.  I will look into slice with append to have O*1) behavior.